### ISSUE TITLE
pywinrm warnings if ansible_winrm_kinit_mode=manual is used

### ISSUE THREAD

**Issue by merlin-tc** [ansible/ansible#23822]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
winrm / kerberos

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Feb 28 2017, 11:34:48) [GCC 5.4.0]
```

##### CONFIGURATION
added option ansible_winrm_kinit_mode=manual to as variable since we use kinit outside ansible.

##### OS / ENVIRONMENT
Gentoo Linux
pywinrm-0.2.2
requests-kerberos-0.11.0

##### SUMMARY
Since the update to ansible 2.3 kerberos doesn't work anymore if there already is a kerberos ticket available.
This can be overridden by setting ansible_winrm_kinit_mode=manual  as mentioned in #20416 . But this keeps producing lots of warnings.
```
 [WARNING]: ansible_winrm_kinit_mode unsupported by pywinrm (is an up-to-date version of pywinrm installed?)
```

##### STEPS TO REPRODUCE

```yaml
ansible server.domain.com -m setup
 [WARNING]: ansible_winrm_kinit_mode unsupported by pywinrm (is an up-to-date version of pywinrm installed?)

server.domain.com | SUCCESS => {
.....
server facts
.....
```

##### EXPECTED RESULTS
If set to manual there shouldn't be any warning that it's unsupported by winrm.

**Comment by nitzmahone** [comment]
It should still work automatically the old way, but only if you didn't specify a password. If a password is specified (by any of the usual means like `ansible_password` inventory var, -u, etc.), then it will default to trying to the new "managed" Kerberos mode.

This is fixed by #24380, and will likely be backported to 2.3.1.

**Comment by nitzmahone** [comment]
Cherry-picked to stable-2.3 for 2.3.1RC1
